docs/guides/connecting_to_your_data/in-memory/pandas.md

```diff
@@ -2,4 +2,145 @@
+
+</Prerequisites>
+
+## Steps
+
+### 1. Choose how to run the code in this guide
+
+<WhereToRunCode />
+
+### 2. `[🍏 CORE SKILL ICON]` Instantiate your project's DataContext
+
+Import these necessary packages and modules.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/in-memory/pandas_yaml_example.py#L1-L6
+```
+
+Load your DataContext into memory using the `get_context()` method.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/in-memory/pandas_yaml_example.py#L8
+```
+
+
+### 3. Configure your Datasource
+
+Using this example configuration we configure a `RuntimeDataConnector` as part of our Datasource, which will take in our in-memory frame.:
+
+<Tabs
+  groupId="yaml-or-python"
+  defaultValue='yaml'
+  values={[
+  {label: 'YAML', value:'yaml'},
+  {label: 'python', value:'python'},
+  ]}>
+  <TabItem value="yaml">
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/in-memory/pandas_yaml_example.py#L10-L22
+```
+
+Run this code to test your configuration.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/in-memory/pandas_yaml_example.py#L24
+```
+
+**Note**: Since the Datasource does not have data passed-in until later, the output will show that no `data_asset_names` are currently available. This is to be expected.
+
+</TabItem>
+<TabItem value="python">
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/in-memory/pandas_python_example.py#L10-L25
+```
+
+Run this code to test your configuration.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/in-memory/pandas_python_example.py#L27
+```
+
+**Note**: Since the Datasource does not have data passed-in until later, the output will show that no `data_asset_names` are currently available. This is to be expected.
+
+</TabItem>
+</Tabs>
+
+
+### 4. Save the Datasource configuration to your DataContext
+
+Save the configuration into your `DataContext` by using the `add_datasource()` function.
+
+<Tabs
+  groupId="yaml-or-python"
+  defaultValue='yaml'
+  values={[
+  {label: 'YAML', value:'yaml'},
+  {label: 'python', value:'python'},
+  ]}>
+  <TabItem value="yaml">
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/in-memory/pandas_yaml_example.py#L26
+```
+
+</TabItem>
+<TabItem value="python">
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/in-memory/pandas_python_example.py#L29
+```
+
+</TabItem>
+</Tabs>
+
+### 5. (Optional) Create Example Data
+
+In most cases you will already have an in-memory dataframe already. For demonstrative purposes, here is a simple dataframe with 3 columns and 3 rows.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/in-memory/pandas_yaml_example.py#L29
+```
+
+Which looks like this:
+
+```bash
+df
+
+   a  b  c
+0  1  2  3
+1  4  5  6
+2  7  8  9
+```
```

Comment:
I think I'd like to push back slightly on this one. Adding the dataframe to the `RuntimeBatchRequest` is the **crucial** step ...  it's just 2 letters "df".  I wonder if we run into the problem of the configuration being too subtle. 

Another option could be to do something similar to data_asset_name and have it be "<YOUR_DATA_FRAME_HERE>", but the risk there is users thinking it has to be the DF passed-in as a string (`"df"`) rather than a variable (`df`)

Reply:
Would you like to talk more about this?